adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala

```diff
@@ -207,7 +210,11 @@ class Transform(protected val args: TransformArgs) extends BDGSparkCommand[Trans
       }
 
       log.info("Sorting reads")
-      adamRecords = oldRdd.sortReadsByReferencePosition()
+      if (args.sortLexicographically) {
+        adamRecords = oldRdd.sortReadsByReferencePosition()
```

Comment:
the pattern here of all these `if` blocks overwriting `adamRecords` is confusing. I know it predates this PR so not necessarily a blocker here, but any thoughts as to how to do all of this more clearly?

e.g. there are a bunch of lines manipulating `adamRecords` above this that are all rendered moot by these lines, seemingly.